src/python/pants/reporting/json_reporter.py

```diff
@@ -0,0 +1,145 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+from builtins import list, open, str
+from collections import defaultdict, namedtuple
+
+from future.moves.itertools import zip_longest
+from six import string_types
+
+from pants.base.build_environment import get_buildroot
+from pants.reporting.reporter import Reporter
+from pants.util.dirutil import safe_mkdir
+
+
+class JsonReporter(Reporter):
+  """Json reporting to files."""
```

Comment:
This docstring mostly just recapitulates the class name. Docstrings and comments should carry their weight. E.g., "A reporter to capture workunit data into a JSON structure." (The fact that it writes to a file is not of immediate interest here).